unittest/server/src/ivi_controller_uinttests.cpp

```diff
@@ -0,0 +1,3475 @@
+ *                         +# get_id_of_layer() must be called once time
+ *                         +# wl_list_insert() must be called once time
+ *                         +# wl_list_init() must be called once time
+ *                         +# get_properties_of_layer() must be called once time
+ *                         +# layer_add_listener() must be called once time
+ *                         +# wl_resource_post_event() must be called {MAX_NUMBER} times
+ *                         +# The result output should same with prepare data
+ *                         +# Free resources are allocated when running the test
+ */
+TEST_F(ControllerTests, layer_event_create)
+{
+    struct ivi_layout_layer l_layoutLayer;
+    l_layoutLayer.id_layer = 10;
+
+    layer_event_create(&mp_iviShell->layer_created, &l_layoutLayer);
+
+    ASSERT_EQ(get_id_of_layer_fake.call_count, 1);
+    ASSERT_EQ(wl_list_insert_fake.call_count, 1);
+    ASSERT_EQ(wl_list_init_fake.call_count, 1);
+    ASSERT_EQ(get_properties_of_layer_fake.call_count, 1);
+    ASSERT_EQ(layer_add_listener_fake.call_count, 1);
+    ASSERT_EQ(wl_resource_post_event_fake.call_count, MAX_NUMBER);
+    struct ivilayer *lp_iviLayer = (struct ivilayer*)(uintptr_t(wl_list_init_fake.arg0_history[0]) - offsetof(struct ivilayer, notification_list));
+    ASSERT_EQ(lp_iviLayer->shell, mp_iviShell);
+    ASSERT_EQ(lp_iviLayer->layout_layer, &l_layoutLayer);
+
+    free(lp_iviLayer);
+}
+
+/** ================================================================================================
+ * @test_id             output_created_event_customScreen
+ * @brief               Test case of output_created_event() where weston output is a custom name
+ *                      different screen info default
+ * @test_procedure Steps:
+ *                      -# Set weston output with name is "custom_screen" and id is 1
+ *                      -# Calling the output_created_event()
+ *                      -# Verification point:
+ *                         +# wl_list_insert() must be called once time
+ *                         +# wl_list_init() must be called once time
+ *                         +# weston_compositor_schedule_repaint() must be called once time
+ *                         +# The result output should same with prepare data
+ *                         +# Free resources are allocated when running the test
+ */
+TEST_F(ControllerTests, output_created_event_customScreen)
+{
+    struct weston_output l_createdOutput;
+    l_createdOutput.id = 1;
+    l_createdOutput.name = (char*)"custom_screen";
+
+    output_created_event(&mp_iviShell->output_created, &l_createdOutput);
+
+    ASSERT_EQ(wl_list_insert_fake.call_count, 1);
+    ASSERT_EQ(wl_list_init_fake.call_count, 1);
+    ASSERT_EQ(weston_compositor_schedule_repaint_fake.call_count, 1);
+    struct iviscreen *lp_iviScreen = (struct iviscreen*)(uintptr_t(wl_list_init_fake.arg0_history[0]) - offsetof(struct iviscreen, resource_list));
```

Comment:
good to get lp_iviScreen pointer via wl_list_insert_fake. care about what is insert to list of screen